src/main/java/org/generationcp/middleware/api/brapi/ObservationServiceBrapiImpl.java

```diff
@@ -0,0 +1,117 @@
+import org.generationcp.middleware.pojos.PhenotypeExternalReference;
+import org.generationcp.middleware.pojos.dms.ExperimentModel;
+import org.generationcp.middleware.pojos.dms.Phenotype;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.CollectionUtils;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.groupingBy;
+
+@Service
+@Transactional
+public class ObservationServiceBrapiImpl implements ObservationServiceBrapi {
+
+	private final DaoFactory daoFactory;
+
+	public ObservationServiceBrapiImpl(final HibernateSessionProvider sessionProvider) {
+		this.daoFactory = new DaoFactory(sessionProvider);
+	}
+
+	@Override
+	public List<ObservationDto> searchObservations(final ObservationSearchRequestDto observationSearchRequestDto, final Integer pageSize,
+		final Integer pageNumber) {
+		final List<ObservationDto> observationDtos = this.daoFactory.getPhenotypeDAO().searchObservations(observationSearchRequestDto, pageSize, pageNumber);
+		if (!CollectionUtils.isEmpty(observationDtos)) {
+			final List<String> observationDbIds = new ArrayList<>(observationDtos.stream().map(ObservationDto::getObservationDbId)
+					.collect(Collectors.toSet()));
+			final Map<String, List<ExternalReferenceDTO>> externalReferencesMap =
+					this.daoFactory.getPhenotypeExternalReferenceDAO().getExternalReferences(observationDbIds).stream()
+							.collect(groupingBy(
+									ExternalReferenceDTO::getEntityId));
+			for(final ObservationDto observationDto: observationDtos) {
+				observationDto.setExternalReferences(externalReferencesMap.get(observationDto.getObservationDbId()));
+			}
+		}
+		return observationDtos;
+	}
+
+	@Override
+	public long countObservations(final ObservationSearchRequestDto observationSearchRequestDto) {
+		return this.daoFactory.getPhenotypeDAO().countObservations(observationSearchRequestDto);
+	}
+
+	@Override
+	public List<Integer> createObservations(final List<ObservationDto> observations) {
+
+		final List<Integer> observationDbIds = new ArrayList<>();
+		final Map<String, ExperimentModel> experimentModelMap = this.daoFactory.getExperimentDao()
+			.getByObsUnitIds(observations.stream().map(ObservationDto::getObservationUnitDbId).collect(Collectors.toList())).stream()
+			.collect(Collectors.toMap(ExperimentModel::getObsUnitId, Function.identity()));
+		final Map<Integer, List<ValueReference>> validValuesForCategoricalVariables =
+			this.getCategoriesForCategoricalVariables(observations);
+
+		final PhenotypeDao dao = this.daoFactory.getPhenotypeDAO();
+		for (final ObservationDto observation : observations) {
+			final Phenotype phenotype = new Phenotype();
+			phenotype.setCreatedDate(new Date());
```

Comment:
Maybe save the `new Date() `to a variable to reuse for both `createdDate` and `updatedDate`.